lib/ui/hooks.dart

```diff
@@ -283,9 +453,16 @@ void _invoke1<A>(void callback(A a)/*?*/, Zone/*!*/ zone, A arg) {
 }
 
 /// Invokes [callback] inside the given [zone] passing it [arg1], [arg2], and [arg3].
-void _invoke3<A1, A2, A3>(void callback(A1 a1, A2 a2, A3 a3)/*?*/, Zone/*!*/ zone, A1 arg1, A2 arg2, A3 arg3) {
-  if (callback == null)
+void _invoke3<A1, A2, A3>(
+  void callback(A1 a1, A2 a2, A3 a3) /*?*/,
+  Zone /*!*/ zone,
```

Comment:
are the spaces before the comments ok? i thought the punctuation had to be literally adjacent to the type